mods/default/chests.lua

```diff
@@ -78,13 +75,49 @@ end)
 
 function default.chest.register_chest(name, d)
 	local def = table.copy(d)
+	local modname, itemname = string.match(name, "(.*):(.*)")
 	def.drawtype = "mesh"
 	def.visual = "mesh"
 	def.paramtype = "light"
 	def.paramtype2 = "facedir"
 	def.legacy_facedir_simple = true
 	def.is_ground_content = false
 
+	-- mod name defaults to 'default:'
+	if modname == nil then
+		minetest.log("warning", "default.chests.register_chest() is " ..
+			"missing the mod name, using 'default:'.")
+		modname = "default"
+		itemname = name
+		name = modname .. ":" .. name
+	end
```

Comment:
This backwards compatibility is not actually needed
For mods _not_ named "default" this function would cause an error anyway. So the only place that could have called this API is the default mod itself.